deepinterpolation/inferrence_collection.py

```diff
@@ -118,6 +128,130 @@ def run(self):
+    except KeyError:
+        model = __load_model_from_mlflow(json_data)
+    return model
+
+
+def core_inference_worker(
+        json_data,
+        input_lookup,
+        rescale,
+        save_raw,
+        output_dict,
+        output_lock):
+
+    model = _load_model(json_data)
+    local_output = {}
+    for dataset_index in input_lookup:
+        local_lookup = input_lookup[dataset_index]
+        local_data = local_lookup['local_data']
+        local_mean = local_lookup['local_mean']
+        local_std = local_lookup['local_std']
+
+        predictions_data = model.predict(local_data[0])
+
+        if rescale:
+            corrected_data = predictions_data * local_std + local_mean
+        else:
+            corrected_data = predictions_data
+
+        corrected_raw = None
+        if save_raw:
+            if rescale:
+                corrected_raw = local_data[1] * local_std + local_mean
+            else:
+                corrected_raw = local_data[1]
+
+        local_output[dataset_index] = {'corrected_raw': corrected_raw,
+                                       'corrected_data': corrected_data}
+
+    with output_lock:
+        k_list = list(local_output.keys())
+        for k in k_list:
+            output_dict[k] = local_output.pop(k)
+    return None
+
+def __get_local_model_path(json_data):
+    try:
+        model_path = json_data['model_path']
+        warnings.warn('Loading model from model_path will be deprecated '
+                      'in a future release')
+    except KeyError:
+        model_path = json_data['model_source']['local_path']
+    return model_path
+
+def __load_local_model(path: str):
+    model = load_model(
+        path,
+        custom_objects={
+            "annealed_loss": lc.loss_selector("annealed_loss")},
+    )
+    return model
+
+def __load_model_from_mlflow(json_data):
+    import mlflow
+    mlflow_registry_params = \
+        json_data['model_source']['mlflow_registry']
+
+    model_name = mlflow_registry_params['model_name']
+    model_version = mlflow_registry_params.get('model_version')
+    model_stage = mlflow_registry_params.get('model_stage')
+
+    mlflow.set_tracking_uri(mlflow_registry_params['tracking_uri'])
+
+    if model_version is not None:
+        model_uri = f"models:/{model_name}/{model_version}"
+    elif model_stage:
+        model_uri = f"models:/{model_name}/{model_stage}"
+    else:
+        # Gets the latest version without any stage
+        model_uri = f"models:/{model_name}/None"
+
+    model = mlflow.keras.load_model(
+        model_uri=model_uri
+    )
+
+    return model
+
+def write_output_to_file(output_dict: Dict,
+                         output_file_path: str,
+                         raw_dataset_name: str,
+                         output_dataset_name: str,
+                         batch_size: int,
+                         first_sample: int) -> Dict:
+    """Function to output results to a h5py file when used in
```

Comment:
Please mention everything this function is doing and give docstring for the parameters
What is the schema for `output_dict` ? Please mention it is writing both corrected_raw as well as corrected_data and under what circumstances it won't write corrected_raw. 

It is appending to data as well as `output_dict` in place